wordpress-mobile/WordPress-iOS
# Prompts: Add Tracks to Bloganuary nudges

Depends on #22129 

As titled, here are the events that are tracked: 

Event | Event Name | Properties
-|-|-
Dashboard card shown | `my_site_dashboard_card_shown` | `type: bloganuary_nudge`, <br>`subtype: bloganuary_nudge`
Contextual menu tapped | `my_site_dashboard_contextual_menu_accessed` | `card: bloganuary_nudge`
Dashboard card hidden | `my_site_dashboard_card_hide_tapped` | `card: bloganuary_nudge`
Learn more button tapped | `bloganuary_nudge_my_site_card_learn_more_tapped` | `prompts_enabled: true, false`
Modal shown | `bloganuary_nudge_learn_more_modal_shown` | `prompts_enabled: true, false`
Modal button tapped | `bloganuary_nudge_learn_more_modal_action_tapped` | `action: turn_prompts_on, dismiss`
Modal dismissed | `bloganuary_nudge_learn_more_modal_dismissed` | n/a

## To test

- Launch the Jetpack app.
- 🔎 Verify that the `my_site_dashboard_card_shown` event is triggered.
- Tap 'Learn more'.
- 🔎 Verify that the `bloganuary_nudge_my_site_card_learn_more_tapped` event is triggered.
- 🔎 Verify that the `bloganuary_nudge_learn_more_modal_shown` event is triggered. 
- Tap the Close button.
- 🔎 Verify that the `bloganuary_nudge_learn_more_modal_dismissed` event is triggered.
- Tap the 'Learn more' button again, and tap the primary button on the modal.
- 🔎 Verify that the `bloganuary_nudge_learn_more_modal_action_tapped` event is triggered. 
- Tap the ellipsis button on the Bloganuary card.
- 🔎 Verify that the `my_site_dashboard_contextual_menu_accessed` event is triggered.
- Tap 'Hide this'. Note: Make sure to do this last (if you plan to verify other variations) since you'd have to reinstall the app to make it appear again.
- 🔎 Verify that the `my_site_dashboard_card_hide_tapped` event is triggered.

## Regression Notes
1. Potential unintended areas of impact
Should be none. Tracking events were added to new components and do not modify existing ones.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested the changes.

3. What automated tests I added (or what prevented me from doing so)
N/A.

PR submission checklist:


UI Changes testing checklist:
N/A
